pkg/runctl/run_manager.go

```diff
@@ -366,7 +366,17 @@ func (c *runManager) setupNetworkPolicyFromConfig(ctx *runContext) error {
 		Kind:  "NetworkPolicy",
 	}
 
-	configStr := c.pipelineRunsConfig.NetworkPolicy
+	configStr := c.pipelineRunsConfig.DefaultNetworkPolicy
+
+	spec := ctx.pipelineRun.GetSpec()
+	if spec.Profiles != nil && spec.Profiles.Network != "" {
+		configStr = c.pipelineRunsConfig.NetworkPolicies[spec.Profiles.Network]
+		if configStr == "" {
+			ctx.pipelineRun.UpdateResult(v1alpha1.ResultErrorConfig)
+			return fmt.Errorf("Cannot find network policy with name %q in network config map", spec.Profiles.Network)
```

Comment:
Do not refer to the internal configuration source of network policies, but just state that the given profile doesn't exist:
```suggestion
			return fmt.Errorf("network profile %q does not exist", spec.Profiles.Network)
```